src/utils/validations/component/validations/with-validation.spec.js

```diff
@@ -0,0 +1,348 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import { withValidation } from '.';
+
+const presErr = new Error('this value is required!');
+const notZeroErr = new Error('this is zero!');
+const asyncErr = new Error('not foo!');
+const failErr = new Error("It's foo!");
+
+const presence = value => new Promise((resolve, reject) => {
+  if (value) {
+    resolve(true);
+  } else {
+    reject(presErr);
+  }
+});
+
+const isNotZero = value => new Promise((resolve, reject) => {
+  if (value !== 0) {
+    resolve(true);
+  } else {
+    reject(notZeroErr);
+  }
+});
+
+const asyncValidation = value => new Promise((resolve, reject) => {
+  setTimeout(() => {
+    if (value === 'foo') {
+      resolve();
+    } else {
+      reject(asyncErr);
+    }
+  }, 200);
+});
+
+const failValidation = val => new Promise((resolve, reject) => {
+  if (val !== 'foo') resolve(true);
+  reject(failErr);
+});
+
+const checkState = (ids, state, res) => {
+  Object.keys(state).forEach((key) => {
+    if (ids.includes(key)) {
+      ids.forEach(id => expect(state[id]).toEqual(res));
+    } else {
+      expect(state[key]).toEqual(!res);
+    }
+  });
+};
+
+const context = { addInput: () => 'foo', removeInput: () => 'foo', adjustCount: () => 'foo' };
+
+const MockComponent = props => <div { ...props } />;
+
+const InputComponent = withValidation(MockComponent);
+
+const WithNoName = withValidation(() => <div />);
+
+const shallowRenderWithContext = props => shallow(<InputComponent { ...props } />, { context });
+
+const validations = [isNotZero, presence, asyncValidation];
+const types = ['validations', 'warnings', 'info'];
+const stateId = ['hasError', 'hasWarning', 'hasInfo'];
+
+describe('Validations component', () => {
+  let wrapper, wrapper2;
+
+  beforeEach(() => {
+    wrapper = shallow(<InputComponent />);
+  });
+
+  it('matches the snapshot when the component renders with default props', () => {
+    wrapper2 = shallowRenderWithContext();
+    expect(wrapper2).toMatchSnapshot();
+  });
+
+  it('matches the snapshot when the hoc is passed an anonymous function with no name or display name', () => {
+    wrapper2 = shallow(<WithNoName />);
+    expect(wrapper2).toMatchSnapshot();
+  });
+
+  describe('renderValidationMarkup', () => {
+    it('returns an Icon with info type when the state has info and no warning or validations', () => {
+      wrapper.setState({ hasInfo: true });
+      checkState([stateId[2]], wrapper.state(), true);
```

Comment:
If I understand correctly: `checkState` method is testing if `hasWarning` state is set to `false`, by setting `true` as it's third argument?  It's a bit counterintuitive IMHO.
I think that `expect(wrapper.state().hasWarning).toEqual(false)` would be more readable.